browser-fetching/index.md

```diff
@@ -0,0 +1,80 @@
+* **Partially updating the page** in response to user interactivity
+  _For example, fetching and displaying the current stock price for a user-provided stock index_
+* **Submitting forms** and displaying a success or error messages without needing to reload an entire page.
+  _For example, Adding an item to a shopping cart_
+
+Fetch's interface only requires a single argument—a `url` address to send the request to. The second argument is an object for specifying various options.
+
+```javascript
+  fetch(url, options) // return a Promise
+```
+As with most asynchronous APIs in JavaScript, `fetch` returns a [Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises) that either resolves (succeeds) or rejects (fails) after receiving a response.
+
+Below is a minimal example of using fetch to request a random dog image from [dog.ceo](https://dog.ceo). The image returned is then displayed in the page along with the full JSON response. 
+
+If you'd like to see the HTTP requests sent during fetches in your browser, open the Web Developer Tools and select the Network tab.
+
+
+```javascript
+const fetchDog = () => fetch('https://dog.ceo/api/breeds/image/random')
+  // Resolve the JSON returned from the url
+  .then(response => response.json())
+  // Display the image and the raw JSON in the page
+  .then(result => {
+    console.log(result)
+    document.querySelector('#dog').src = result.message
+    document.querySelector('#dog-results').innerHTML = JSON.stringify(result, null, 2)
+  })
+  // Log any errors to the developer console
+  .catch(error => console.error(error))
+```
+
+<script>
+const fetchDog = () => fetch('https://dog.ceo/api/breeds/image/random')
+  .then(response => response.json())
+  .then(result => {
+    console.log(result)
+    document.querySelector('#dog').src = result.message
+    document.querySelector('#dog-results').innerHTML = JSON.stringify(result, null, 2)
+  })
+  .catch(error => console.error(error))
+</script>
+<button onclick="fetchDog()">Call the `fetchDog` function</button>
+
+#### JSON from the response:
+<pre class="language-javascript" id="dog-results">Nothing fetched yet.</pre>
+
+#### Image from the response:
+<div style="width: 400px;">
+  <img id="dog" style="width: 100%;" alt="Random dog image" src="https://via.placeholder.com/400" />
+</div>
+
+Below is more complex example that uses multiple options to `POST` a request to a weather API:
+
+```
+TODO
+```
+
+## Summary
+
+The previous examples only hint at the capabilities of the Fetch API. Read the ["Using the Fetch API"](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch) page on the Mozilla Developer Network (MDN) for more comprehensive examples and detail on the Fetch API. Fetch isn't limited to web browsers—it's also available in [Node.js v18](https://nodejs.org/en/blog/announcements/v18-release-announce/#fetch-experimental) and can be used on the server.
```

Comment:
We need to keep in mind context. Let's just stipulate that our outreach program has both adults and younger learners, who are unfamiliar with tech. So, we have to somehow balance this goal of catering to these learners with different inclinations with our writing/visual effects.

I thought of a useful idea to help us handle this situation, and appeal to a broader audience.

**THE KID BOX**

Think of this, it's like a visual box, with some additional information. It's shiny and colorful, and the idea is this.

A kid comes to site. They are on the learning module page. Although they're following the same module as the adult learner, there are these boxes that say something like "Kids only!" This will do a few things for us. It will allow us to add optional information that people can look at if they want (and it may be helpful to some adults, too, we don't know, so we cast a wide net). These boxes help to augment the kid's understanding, and now we have a flexible mechanism that allows us to hopefully tailor our explanation to target audiences that are distinct in important ways, in terms of how they approach learning.

Reply:
This also allows us to not force our simple and clear content to be juvenile and not appeal to kids. if kids need more info, put it in the kid box. But the kid box is also more versatile than that. It can be used not only to provide information for kids, but to do something else with language or visual affects. What? Sometimes a piece of language can be used to entertain, for example, and that can be useful.

Reply:
This also helps with our branding. Parents love kids content. They'd be more receptive to our site if they see it as something for kids. And, this in turn, plays into our long term gain of securing more donations and gaining more exposure so that we can get money, and pursue the Free Free API Education company's goal to the best of our ability.

Reply:
Actually, I think we want two kinds of supplementary information boxes. Perspective and Kids.

We can think about the exact styling for these later. What we want to do first is evaluate them from a structural and educational perspective.

Use the Perspective BOX (name pending) to provide perspective wherever it is warranted. Now we have a standard mechanisms that everyone understands how to use. And we can keep the styles centralized for this in one place, which will make it super handy to just focus on content.

Reply:
And, for example, if the function of a piece of writing is to persuade. Sometimes that will be the case. You don't necessarily have to write it, if you feel more comfortable with someone else doing that kind of writing. For example, you could be like "hey, team member X, can I get a persuasive paragraph on this topic? I'm pretty good with simple and clear style but that might not mean im the best persuasive writer, and that's okay. Use team's resources.